src/highlights/v0.24.md

```diff
@@ -0,0 +1,300 @@
+
+All the code is available at <https://github.com/ratatui-org/ratatui-book>
+
+## Frame: no more generics 🚫
+
+If you were using the `Frame` type with a `Backend` parameter before:
+
+```rs
+fn draw<B: Backend>(frame: &mut Frame<B>) {
+    // ...
+}
+```
+
+You no longer need to provide a generic over backend (`B`):
+
+```rs
+fn draw(frame: &mut Frame) {
+    // ...
+}
+```
+
+## New Demo / Examples ✨
+
+We have a new kick-ass demo!
+
+![demo2](https://raw.githubusercontent.com/ratatui-org/ratatui/aa09e59dc0058347f68d7c1e0c91f863c6f2b8c9/examples/demo2.gif)
+
+To try it out:
+
+```sh
+cargo run --example demo2
+```
+
+> The code is available [here](https://github.com/ratatui-org/ratatui/tree/main/examples/demo2).
+
+We also have a new example demonstrating how to create a custom widget.
+
+![custom widget](https://raw.githubusercontent.com/ratatui-org/ratatui/images/examples/custom_widget.gif)
+
+```sh
+cargo run --example=custom_widget --features=crossterm
+```
+
+> The code is available [here](https://github.com/ratatui-org/ratatui/blob/main/examples/custom_widget.rs).
+
+Lastly, we added an example to demonstrate RGB color options:
+
+![RGB colors](https://raw.githubusercontent.com/ratatui-org/ratatui/images/examples/colors_rgb.gif)
+
+```sh
+cargo run --example=colors_rgb --features=crossterm
+```
+
+> The code is available [here](https://github.com/ratatui-org/ratatui/blob/main/examples/colors_rgb.rs).
+
+## Window Size API 🪟
+
+A new method called `window_size` is added for retrieving the window size.
+It returns a struct called `WindowSize` that contains both pixels (width, height) and columns/rows
+information.
+
+```rs
+let stdout = std::io::stdout();
+let mut backend = CrosstermBackend::new(stdout);
+println!("{:#?}", backend.window_size().unwrap());
+```
+
+Outputs:
+
+```
+WindowSize {
+    columns_rows: Size {
+        width: 240,
+        height: 30,
+    },
+    pixels: Size {
+        width: 0,
+        height: 0,
+    },
+}
+```
+
+With this new API, the goal is to improve image handling in terminal emulators
+by sharing terminal size and layout information,
+enabling precise image placement and resizing within rectangles.
+
+See the pull request for more information:
+[https://github.com/ratatui-org/ratatui/pull/276](https://github.com/ratatui-org/ratatui/pull/276)
+
+## BarChart: Render _smol_ charts 📊
+
+We had a bug where the `BarChart` widget doesn't render labels that are full width.
+Now this is fixed and we are able to render charts smaller than 3 lines!
+
+For example, here is how `BarChart` is rendered and resized from a single line to 4 lines in order:
+
+```
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+      Group
+
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
+
+If you set `bar_width(2)` then it is rendered as:
+
+```
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
```

Comment:
I'm not sure that the example is correct.

Reply:
Should be more clear now: ef4d9a46ee916bebd28cfc9b6425f6750d18f36b